### PR TITLE
Feature/get token promise support

### PR DESCRIPTION
**Pull request by mlabrum**
Hey,

Thanks for this project :D

This adds support for promises within `getToken` (if using default network helpers) and `NetworkHelpers` methods are now allowed to return a `Promise`

I've wrapped the return values of both `getToken` and the network helpers in a `Promise.resolve` to ensure this works in a backwards compatible way as any value returned from them will now be resolved into a promise and `await`ed.

Cheers,

### CODE REVIEW

File: src/internals/network/getDefaultNetworkHelpers.js [hunk 1]
```diff
@@ -1,62 +1,72 @@
 const DEFAULT_NETWORK_HELPERS = {
   getToken: () => 'token',
-  requestGET() {
+  async requestGET() {
+    const token = await Promise.resolve(this.getToken && this.getToken())
```
Comment (Zephir77167) [inline src/internals/network/getDefaultNetworkHelpers.js]: I don't understand the need for `Promise.resolve` here, wouldn't `const token = this.getToken && await this.getToken()` work the same?

File: __tests__/internals/network/networkHelpers.test.js [hunk 2]
```diff
@@ -8,16 +8,31 @@ describe('getNetworkHelpers', () => {
...
       `Bearer ${tokenName}`,
     );
   });
+
+
+  test('after setting them - promise token', async () => {
+    const tokenName = 'customToken';
+
+    setNetworkHelpers({
+      getToken: () => Promise.resolve(tokenName),
```
Comment (Zephir77167) [inline __tests__/internals/network/networkHelpers.test.js]: I think this line could be simplified to `getToken: async () => tokenName,` to highlight the asynchronicity of `getToken`

File: src/internals/network/getDefaultNetworkHelpers.js (hunk 1 above)
Comment (mlabrum) [reply src/internals/network/getDefaultNetworkHelpers.js]: Hey, good catch I didn't realize that await would turn the value into a promise, I thought it only worked on promises, that's why I had the promise.resolve :D I'll update this

**Comment by mlabrum** [comment]
I've updated the PR, I didn't realize that `await` automatically wraps the value into a `Promise`

File: src/internals/action-creator/thunk/generateActionCreator.js [hunk 3]
```diff
@@ -39,7 +39,7 @@ const generateActionCreator = (
 
     const res = await fetch(
       formattedURL,
-      combinedNetworkHelpers[`request${method}`](finalBody),
+      (await Promise.resolve(combinedNetworkHelpers[`request${method}`](finalBody))),
```
Comment (Zephir77167) [inline src/internals/action-creator/thunk/generateActionCreator.js]: Can we get rid of this one as well?

**Comment by mlabrum** [comment]
Ah missed that one, I'll get that fixed :)

**Comment by Zephir77167** [comment]
Perfect, thanks a lot for the PR!

**Pull request merged**